src/common/arrstr.cpp

```diff
@@ -573,3 +575,170 @@ wxArrayString wxSplit(const wxString& str, const wxChar sep, const wxChar escape
 
     return ret;
 }
+
+
+#ifdef wxUSE_REGEX
+
+wxRegEx naturalNumeric(_T("[0-9.]+"));
+wxRegEx naturalAlpha(_T("[^0-9.]+"));
+
+enum wxStringFragmentType
+{
+    wxFRAGMENT_TYPE_EMPTY = 0,
+    wxFRAGMENT_TYPE_ALPHA = 1,
+    wxFRAGMENT_TYPE_DIGIT = 2
+};
+
+// ----------------------------------------------------------------------------
+// wxStringFragment
+// ----------------------------------------------------------------------------
+// 
+// Lightweight object returned by wxString::GetNaturalFragment()
+// Consists of either a string which represents a number, or a string which contains no numerical digits.
+// Is used by wxString::CompareNatural()
+class wxStringFragment
+{
+public:
+    wxStringFragment()
+        : type(wxFRAGMENT_TYPE_EMPTY)
+    {}
+
+    wxString             text;
+    double               value;
+    wxStringFragmentType type;
+};
+
+
+wxStringFragment GetFragment(wxString &text)
+{
+    size_t digitStart = 0;
+    size_t digitLength = 0;
+    size_t alphaStart = 0;
+    size_t alphaLength = 0;
+    wxStringFragment fragment;
+
+    if (text.Length() == 0)
```

Comment:
Please use `std::string`-compatible methods in the new code, i.e. `length()` instead of `Length()`.

Also, here it's not needed at all, just write the test as `if ( text.empty() )`.